packages/@aws-cdk/integ-runner/lib/runner/integration-tests.ts

```diff
@@ -190,6 +233,11 @@ export interface IntegrationTestFileConfig extends IntegrationTestsDiscoveryOpti
  * Discover integration tests
  */
 export class IntegrationTests {
+  private static readonly defaultDiscoveryRegexes = new Map<string, RegExp>([
+    ['javascript', new RegExp(/^integ\..*\.js$/)],
+    // Allow files ending in .ts but not in .d.ts
+    ['typescript', new RegExp(/^integ\..*(?<!\.d)\.ts$/)],
+  ]);
```

Comment:
I see why you had to split the regexes and app commands. But the split and duplication with language detection feels odd.